src/components/item/PinModal.vue

```diff
@@ -0,0 +1,302 @@
+      >
+        {{ selectedIsPinned ? $t('entity.actions.unpin') : $t('entity.actions.pin') }}
+      </b-button>
+      <span
+        v-else-if="selectedIsFull"
+      >
+        <b-button
+          data-qa="go to set link"
+          :to="$path(selectedLink)"
+          variant="primary"
+        >
+          {{ $t('entity.actions.viewPinned') }}
+        </b-button>
+      </span>
+    </div>
+  </b-modal>
+</template>
+
+<script>
+  import pick from 'lodash/pick';
+  import { mapGetters, mapState } from 'vuex';
+  import makeToastMixin from '@/mixins/makeToast';
+
+  export default {
+    name: 'PinModal',
+
+    mixins: [
+      makeToastMixin
+    ],
+
+    props: {
+      modalId: {
+        type: String,
+        default: 'pin-modal'
+      },
+      entities: {
+        type: Array,
+        required: true
+      },
+      /**
+       * Used for testing, in order to render the modal.
+       */
+      modalStatic: {
+        type: Boolean,
+        default: false
+      }
+    },
+    data() {
+      return {
+        fetched: false,
+        selected: null
+      };
+    },
+
+    computed: {
+      infoText() {
+        if (this.selected) {
+          if (this.selectedIsFull) {
+            return this.selectedIsPinned ? this.$t('entity.notifications.unpin', { entity: this.selectedEntityPrefLabel }) : this.$t('entity.notifications.pinLimit.body');
+          }
+          if (this.selectedIsPinned) {
+            return this.$t('entity.notifications.unpin', { entity: this.selectedEntityPrefLabel });
+          }
+          return this.$t('entity.notifications.pin', { entity: this.selectedEntityPrefLabel });
+        }
+        return this.$t('entity.notifications.select');
+      },
+      selectedIsPinned() {
+        return this.selected && this.pinnedTo(this.selected);
+      },
+      selectedIsFull() {
+        return this.selected && this.featuredSetPins[this.selected]?.length >= 24;
+      },
+      selectedLink() {
+        return { name: 'set-all', params: { pathMatch: this.selected && this.featuredSetIds[this.selected].replace('http://data.europeana.eu/set/', '') } };
+      },
+      selectedEntityPrefLabel() {
+        return this.allRelatedEntities.find(entity => entity.id === this.selected)?.prefLabel?.en;
+      },
+      ...mapGetters({
+        itemId: 'item/id',
+        pinnedTo: 'item/pinnedTo'
+      }),
+      ...mapState({
+        featuredSetPins: state => state.item.featuredSetPins,
+        featuredSetIds: state => state.item.featuredSetIds,
+        allRelatedEntities: state => state.item.allRelatedEntities
+      })
+    },
+
+    methods: {
+      fetchPinningData() {
+        // TODO: Don't fetch all entities if related entities are already present and less than 5. Set all entities to related entities instead for that scenario.
+        return this.$apis.entity.find(this.entities)
+          .then(entities => entities.map(entity => pick(entity, ['id', 'prefLabel', 'isShownBy', 'logo'])))
```

Comment:
If this is only used for pinning, does it need the logo for example?

Reply:
well the original design was also including background images and logo/depiction icons.
So that's why it's being retrieved here, but yes it's not being used at the moment.